app/validators/questionnaire_validator.py

```diff
@@ -140,6 +142,118 @@ def validate_introduction_block(self):
+        for group in self.questionnaire_schema.groups:
+            self.validate_answer_source_group(group)
+
+        # Handling section level "enabled" rule
+        for index, section in enumerate(self.questionnaire_schema.sections):
+            self.validate_answer_source_section(section, index)
+
+    def validate_answer_source_group(self, group):
+        identifier_references = get_object_containing_key(group, "source")
+        for path, identifier_reference, parent_block in identifier_references:
+            # set up default parent_block_id for later check (group or block level)
+            parent_block_id = None
+            if (
+                "source" in identifier_reference
+                and identifier_reference["source"] == "answers"
+            ):
+
+                source_block = self.questionnaire_schema.get_block_by_answer_id(
+                    identifier_reference["identifier"]
+                )
+                # Handling non-existing blocks used as source
+                if not source_block:
+                    self.add_error(
+                        ValueSourceValidator.ANSWER_SOURCE_REFERENCE_INVALID,
+                        identifier=identifier_reference["identifier"],
+                    )
+                    return False
+                # Handling block level answer sources (skipping group level)
+                if parent_block and "blocks" in path:
+                    parent_block_id = parent_block["id"]
+                    parent_block_index = self.questionnaire_schema.block_ids.index(
+                        parent_block_id
+                    )
+                else:
+                    # Handling group level skip conditions
+                    first_block_id_in_group = group["blocks"][0]["id"]
+                    parent_block_index = self.questionnaire_schema.block_ids.index(
+                        first_block_id_in_group
+                    )
+
+                source_block_id = self.resolve_source_block_id(source_block)
+
+                source_block_index = self.questionnaire_schema.block_ids.index(
+                    source_block_id
+                )
+                if source_block_index > parent_block_index:
+                    if parent_block_id:
+                        self.add_error(
+                            error_messages.ANSWER_REFERENCED_BEFORE_EXISTS.format(
+                                answer_id=identifier_reference["identifier"]
+                            ),
+                            block_id=parent_block_id,
+                        )
+                    else:
+                        self.add_error(
+                            error_messages.ANSWER_REFERENCED_BEFORE_EXISTS.format(
+                                answer_id=identifier_reference["identifier"]
+                            ),
+                            group_id=group["id"],
+                        )
+
+    def validate_answer_source_section(self, section, section_index):
```

Comment:
I might be looking in the wrong place but would this need a new test? Or is it okay for coverage?

Reply:
Sort of same as: https://github.com/ONSdigital/eq-questionnaire-validator/pull/245#discussion_r1677464858